spec/components/grid-item-spec.js

```diff
@@ -0,0 +1,81 @@
+import createComponent from '../helpers/createComponent.js';
+import GridItem, { getFormattedName } from '../../src/renderer/components/grid-item.jsx';
+import generatorsList from '../fixtures/generators-list.js';
+
+
+function mockContent() {
+  const contentDiv = window.document.createElement('div');
+  contentDiv.id = 'content';
+  contentDiv.style.height = '800px';
+  window.document.body.appendChild(contentDiv);
+}
+
+function dismissContent() {
+  window.document.body.removeChild(
+    window.document.getElementById('content')
+  );
+}
+
+describe('Grid Item', () => {
+  let renderOutput;
+
+  beforeEach(() => {
+    mockContent();
+    renderOutput = createComponent(
+      GridItem,
+      generatorsList[0]
+    );
+  });
+
+  afterEach(dismissContent);
+
+  it('should build container element', () => {
+    expect(
+      renderOutput.props.children[0].type
+    ).toBe(
+      'div'
+    );
+  });
+
+  it('should set callback functions', () => {
+    [
+      'onMouseOver',
+      'onMouseOut',
+      'onClick'
+    ].forEach(fnName => {
+      expect(
+        typeof renderOutput.props[fnName]
+      ).toBe(
+        'function'
+      );
+    });
+  });
+
+  it('should have a background element', () => {
+    expect(
+      renderOutput.props.children[1].type
+    ).toBe(
+      'figure'
+    );
+    expect(
+      renderOutput.props.children[1].props.style.backgroundImage
+    ).toBeTruthy();
+  });
+
+
+  it('should have a title element', () => {
+    expect(
+      renderOutput.props.children[2].type
+    ).toBe(
+      'h3'
+    );
+  });
+
+  it('should show correct grid item name', () => {
+    expect(
+      renderOutput.props.children[2].props.children
+    ).toBe(
+      getFormattedName(generatorsList[0].name)
```

Comment:
This one got me really intrigued, how would be the best way to not have to expose `getFormattedName`?

Are you suggesting that this transformation should be made at another level? or that I should just duplicate the function in the test case? I have no problems exposing methods for the sake of tests.